docs/architecture.md

```diff
@@ -0,0 +1,138 @@
+new TEE-based VSIs on demand for the pod sandboxes and we can create a network tunnel between the node and sandbox for
+commands to flow.
+
+There are some limitations and advantages of this approach including:
+
+#### Limitations
+
+In the current Kata Containers approach, the container runtime creates a pod sandbox as a VM inside the Kubernetes node
+using a local hypervisor and then uses a local virtual network to communicate to it. Compared to this existing approach,
+using CSP IaaS APIs to create a new VM per pod sandbox adds an overhead in time and resources. This means the peer pods
+solution probably won't work well for small, short-lived applications.
+Using a virtual network to communicate between the node (host) and the pod sandbox (guest) components, the approach of
+peer pods is more complicated as well
+
+#### Advantages
+Due to the lack of wide availability of running TEEs in nested VM, if we want to use TEEs, then compared to having a bare
+metal machine per node, we can scale more easily and cheaply.
+There is also some advantages to the independence of peer pods, where each pod sandbox has it's own dedicated resources
+like network and storage, rather than sharing these with the Kubernetes node and other pods. This can help to improve
+resource management, reliability as well as security.
+
+## Components
+
+The diagram below shows the components that are installed into a Kubernetes cluster when Peer pods is enabled
+
+<!--the architecture diagram needs updating with the extra components Issue #1291-->
+![Architecture](./architecture.png)
+
+<!-- TODO - do we want to add lots of info here, or point to READMEs in the different components' directories?-->
+
+### `containerd-shim-kata-v2`
+
+The [`shim v2`](https://github.com/kata-containers/kata-containers/blob/main/docs/design/architecture/README.md#shim-v2-architecture)
+part of the Kata Containers runtime, implements the containerd runtime v2 API for Kata Containers and allows it to
+receive commands from Kubernetes, via a CRI client to launch and run Pods.
+
+
+### Cloud API Adaptor
+The [cloud-api-adaptor](../cmd/cloud-api-adaptor) implements the remote hypervisor. It runs as a daemonset on
+each node and is responsible for receiving commands from the `containerd-shim-kata-v2 process` and implementing them
+for peer pods. More example, when it recieved commands that were related to Pod lifecycle that are traditionally
+serviced by a local hypervisor (like `CreateSandboxRequest` and `StopSandboxRequest`) it will use IaaS APIs (from the
+Cloud Service Provider, or locally for the `libvirt` implementation to creating and deleting the peer pod VMs. 
+
+When the VMs are created they make a CNI compatible network tunnel using VxLAN tunneling, between the worker node and
+peer pods VM to flow other commands like `CreateContainer` through to the the remote sandbox.
+
+### Agent Protocol Forwarder
```

Comment:
Worth having a section before this that we create our own cloud images for each provider with these components already present? And then direct to the podvm directory.

Reply:
Thanks for the suggestion - I've separated the components into cluster node and VM and added a small intro and link.